File: src/main/python/lab999_functions/yearApp.py
"""
 year function.

 @author rambabu.posa
"""
from pyspark.sql import SparkSession
from pyspark.sql import (functions as F)
import os

current_dir = os.path.dirname(__file__)
relative_path = "../../../../data/functions/dates.csv"
absolute_file_path = os.path.join(current_dir, relative_path)

# Creates a session on a local master
spark = SparkSession.builder.appName("year function") \
    .master("local[*]").getOrCreate()

df = spark.read.format("csv") \
    .option("header", True) \
    .option("imferSchema", True) \
    .load(absolute_file_path)

df = df.withColumn("year", F.year(F.col("date_time")))

df.show(5, False)
df.printSchema()

spark.stop()

File: src/main/java/net/jgp/books/spark/ch13/lab999_functions/YearApp.java
package net.jgp.books.spark.ch13.lab999_functions;

import static org.apache.spark.sql.functions.col;
import static org.apache.spark.sql.functions.year;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

/**
 * year function.
 * 
 * @author jgp
 */
public class YearApp {

  public static void main(String[] args) {
    YearApp app = new YearApp();
    app.start();
  }

  /**
   * The processing code.
   */
  private void start() {
    SparkSession spark = SparkSession.builder()
        .appName("year function")
        .master("local[*]")
        .getOrCreate();

    Dataset<Row> df = spark.read().format("csv")
        .option("header", true)
        .option("imferSchema", true)
        .load("data/functions/dates.csv");

    df = df.withColumn("year", year(col("date_time")));

    df.show(5, false);
    df.printSchema();
  }
}
